libs/utils/trace_analysis.py

```diff
@@ -987,8 +1009,9 @@ def getClusterFrequencyResidency(self, cluster):
         #     freq_active[t] == 1 if at time t the frequency is f
         #     freq_active[t] == 0 otherwise
         available_freqs = sorted(cluster_freqs.unique())
-        new_idx = sorted(cluster_freqs.index.tolist() + \
-                         cluster_active.index.tolist())
+        # Use set() to remove possible duplicate indexes
+        new_idx = sorted(set(cluster_freqs.index.tolist() + \
```

Comment:
Since these two series are generated from two different trace points, I would say it can never happen that we have two equal indexes in these two series. Maybe this set() is not required, or otherwise we need to add either a better comment or raise an exception for a malformed trace file.